google/cloud/bigtable/data/_async/mutations_batcher.py

```diff
@@ -305,7 +305,7 @@ async def append(self, mutation_entry: RowMutationEntry):
         # TODO: return a future to track completion of this entry
         if self._closed.is_set():
             raise RuntimeError("Cannot append to closed MutationsBatcher")
-        if isinstance(mutation_entry, Mutation):  # type: ignore
+        if isinstance(cast(Mutation, mutation_entry), Mutation):
```

Comment:
TL;DR: because comments aren't part of the ast, so `# type: ignore` doesn't make it through the generation process for the sync client. But `typing.cast` does the same thing, by forcing the type checker to treat it as a specific type here

For why we need to ignore the type in the first place: this is a check to raise a helpful error when the user passes in a `Mutation` instead of a `RowMutationEntry`, which could be an easy mistake to make. But the type checker doesn't like this check, since we're checking if it's a type that it knows it should never be

Reply:
Thanks for the explanation!